BookClub/tests/models/test_add_user_club_model.py

```diff
@@ -0,0 +1,87 @@
+
+
+from ssl import MemoryBIO
+from django.forms import ValidationError
+from django.test import TestCase
+from BookClub.models import Club, User, ClubMembership
+
+class AddUserTestCase(TestCase):
+    
+    fixtures = [
+        'BookClub/tests/fixtures/default_users.json',
+        'BookClub/tests/fixtures/default_clubs.json',
+        'BookClub/tests/fixtures/default_club_owners.json',
+        ]
+
+    def setUp(self):
+        self.club1 = Club.objects.get(pk = 1)
+        self.club2 = Club.objects.get(pk = 2)
+        self.user1 = User.objects.get(pk=1)
+        self.user2 = User.objects.get(pk=2)
+        self.testUser = User.objects.create(
+            username="Dave",
+            email="davestation@davestation.com",
+            public_bio = "I am dave yognaut and I have the pants",
+            password="Password123",
+        )
+    def _assert_club_is_valid(self):
+        try:
+            self.club1.full_clean()
+        except(ValidationError):
+            self.fail('Test club should be valid')
+
+    def _assert_club_is_invalid(self):
+        with self.assertRaises(ValidationError):
+            self.club1.full_clean()
+
+    def test_valid_club(self):
+        self._assert_club_is_valid()
+        
+    def testAddUserValidInfo(self):
+        membershipBefore = ClubMembership.objects.filter(club=self.club1).count()
+        self.club1.add_user(self.testUser, ClubMembership.UserRoles.MEMBER)
+        membershipAfter = ClubMembership.objects.filter(club=self.club1).count()
+        self.assertLess(membershipBefore, membershipAfter)
```

Comment:
For all tests using assertLess(), it's better to use assertEquals with before + 1 for better accuracy

Reply:
Solved